### PR TITLE
Add `cargo uniffi-bindgen` alias for running uniffi-bindgen.

### PR DESCRIPTION
**Pull request by rfk**
Whelp, I've had about as many uniffi-version-mismatch-related build failures as I'm willing to tolerate. I think the requirement to have `uniffi-bindgen` installed as a separate command on the user's machine was a great way to get started quickly, but it's making it hard to iterate on our tooling, particularly when trying to co-develop features in uniffi with a consumer over here in application-services.

Inspired by how we handle protobuf files, here's a proposal for an alternative - `cargo uniffi-bindgen` alias that works in all respects like the system-level `uniffi-bindgen` command, but which always uses the version of uniffi that's currently in use by our workspace. I expect this should completely eliminate the possibility of UniFFI version mismatch errors (with the possible exception of failing to clean up files generated by a previous version).

One downside of this change, is that our compile times will get slightly longer, since you have to build `uniffi_bindgen` in the workspace rather than using the one that's already built on your system. I'll take that trade in the interests of reducing failure modes, but YMMV. (I'll add that it shouldn't really affect CI times since we `cargo install uniffi_bindgen` as part of our CI scripts anyway).

This is a draft until we get a release of UniFFI with https://github.com/mozilla/uniffi-rs/pull/406 included, but pushing it early for feedback.

### CODE REVIEW

File: Cargo.toml
Comment (mhammond) [inline Cargo.toml:43]: 'builtin' kinda jarred with me - 'embedded' sounds a bit better, but I'm not sure why as that's pretty much a synonym. I think 'builtin' made me thing 'built in to uniffi'. No strong opinion here, but I thought I'd open a bikeshed.

**Review by mhammond — COMMENTED** [review]
+1 to the approach!

File: Cargo.toml
Comment (rfk) [reply Cargo.toml:43]: Fair point. I invested about five seconds worth of effort into the name here, and went with "builtin" entirely by analogy to `features=["builtin-bindgen"]` which you need to add to the config of each crate. I'd be fine with "embedded", and it might also help to re-order to words to something like "embedded-uniffi-bindgen" or "builtin-uniffi-bindgen" so that it's less suggestive of being built in to uniffi.

**Comment by rfk** [comment]
Since this is a non-trivial workflow change, I'd like to get feedback from both current and potential future contributors to this repo; flagging @lougeniaC64 @dmose @travis79 @jhugman for visibility, please take a look and leave any thoughts at your convenience.

**Comment by rfk** [comment]
After https://github.com/mozilla/application-services/pull/3890, I think we need this sooner rather than later, so I've made a new uniffi release and am moving this into "ready for review". (Still want to give everyone a change to weigh in though).

File: components/autofill/android/build.gradle
Comment (rfk) [inline components/autofill/android/build.gradle:106]: Unfortunately `cargo uniffi-bindgen` doesn't work within a crate directory, only in the workspace root. We can still call it from within gradle in a reasonable fashion though.

**Comment by rfk** [comment]
One alternative here, which @jdragojevic asked me about the other day: we could consider moving `uniffi-rs` development into this repo like we used to do for `ffi-support`. I'm lightly -1 on the idea because I think it might discourage other projects (both internal and external) from using it, but am open to having my mind changed. And I think we'd still want to set up a `cargo uniffi-bindgen` alias for simplicity.

**Comment by lougeniaC64** [comment]
> Since this is a non-trivial workflow change, I'd like to get feedback from both current and potential future contributors to this repo; flagging @lougeniaC64 @dmose @travis79 @jhugman for visibility, please take a look and leave any thoughts at your convenience.

I've had enough experience setting my local version of `uniffi-bindgen` to warmly welcome slightly longer compile times. 🚀

**Comment by dmose** [comment]
Seems high-value to me.  I'm also pretty much perfectly aligned with your analysis on pulling uniffi-rs into this repo.

On a compile-time-related note, using sccache via RUSTC_WRAPPER to build this repo cut the build time after a cargo clean from ~93s to ~49s wall-clock time on my Macbook Pro.

**Review by travis79 — APPROVED** [review]
+1 from me on this in general.  I can't count the number of times I've used `cargo install uniffi_bindgen --force` recently.  I'm also leaning away from pulling uniffi-rs into A-S completely, but I suppose it could be set up in such a way as to publish to crates.io and still be easily consumable by outside consumers.

**Comment by rfk** [comment]
> I'm also leaning away from pulling uniffi-rs into A-S completely, but I suppose it could be set up
> in such a way as to publish to crates.io and still be easily consumable by outside consumers.

Indeed. So for a bit of historical context, we used to have the `ffi-support` crate developed in this repo but published to crates.io. This is the bug in which we discussed moving it into a separate repo: https://github.com/mozilla/application-services/issues/3463

I think either approach could work, but having it in a separate repo may be slightly less friction overall.

**Comment by rfk** [comment]
It's not clear to me why Travis's r+ isn't enough to unblock merging here, but if one of @mhammond or @lougeniaC64 could please r+ this as well, that would be awesome :-)

**Comment by rfk** [comment]
> It's not clear to me why Travis's r+ isn't enough to unblock merging here, 

Aha! I added Travis to a team with write access to this repo and that seems to have done the trick.

**Comment by rfk** [comment]
James hasn't weighed in but:

* I'm hearing strong support in the discussion here.
* This prevents an actual footgun that just shot me in the foot yesterday.
* This doesn't require any action from developers after landing.
* It's easy to undo this change if we decide we don't like it.

So I'm going to go ahead and merge.

**Pull request merged**
